_includes/devdoc/ref_p2p_networking.md

```diff
@@ -167,6 +168,88 @@ different reasons:
+
+{% autocrossref %}
+
+*Added in protocol version 70014 as described by BIP152.*
+
+A `cmpctblock` message is a compact enocoding of a new block, used to relay
+blocks with faster propogation and lower bandwidth than relaying the full
+block. See `sendcmpct` for details of compact block relay modes.
+
+`cmpctblock` contains the block header, short transaction ids for the
+transactions which the node expects the peer to already have, and
+PrefilledTransaction structures for the other transactions in the block
+(which the node expects the peer to not yet have).
+
+| Bytes                   | Name                 | Data Type                     | Description
+|-------------------------|----------------------|-------------------------------|----------------
+| 80                      | header               | Block header                  | The header of the block being provided. The first 80 bytes of the block as defined by the encoding used by "block" messages.
+| 8                       | nonce                | uint64_t                      | A nonce for use in short transaction ID calculations.
+| *Varies* (1 or 3 bytes) | shortids_length      | compactSize uint              | The number of short transaction IDs (ie block tx count - prefilledtxn_length).
+| *Varies*                | shortids             | List of 6-byte integers       | The short transaction IDs calculated from the transactions which are not provided explicitly in prefilledtxn. The construction of the short transaction ID is described below.
+| *Varies* (1 or 3 bytes) | prefilledtxn_length  | compactSize uint              | The number of prefilled transactions in prefilledtxn (ie block tx count - shortids_length).
+| *Varies*                | prefilledtxn         | List of PrefilledTransactions | The transactions which the node expects the peer to be missing. Always includes the coinbase transaction and may include additional transactions in the block. The format of the PrefilledTransaction structure is defined below.
+
+The Prefilled Transaction structure is:
+
+| Bytes             | Name        | Data Type        | Description
+|-------------------|-------------|------------------|----------------
+| *Varies* (1 or 3) | index       | compactSize uint | The index of the transaction in the block, differentially encoded.
+| *Varies*          | Transaction | Transaction      | The transaction being relayed in raw transaction form.
+
+The indexes are *differentially encoded*. Instead of using raw indexes, the
+number encoded is the difference between the current index and the previous
+index, minus one. For example, a first index of 0 implies a real index of 0,
+a second index of 0 thereafter refers to a real index of 1, etc.
+
+Short transaction IDs are used as a compact representation of transaction to
+relay without sending a full 256-bit hash. They are calculated by:
+
+1. Hashing `block header||nonce` (in little-endian) with a single-SHA256 hash
```

Comment:
The Travis CI build error is because our auto-crossref tool is trying to link `block header` to our glossary entry for block header.  You can resolve this error by changing the text above to:

    `block<!--noref--> header<!--noref-->||nonce`

(Ugly, I know, but it works.)